docs/create_your_own_brick.rst

```diff
@@ -0,0 +1,371 @@
+    ...             The two inputs multiplied by their respective matrices
+    ...
+    ...         """
+    ...         W1, W2 = self.parameters
+    ...         output1 = tensor.dot(input1_, W1)
+    ...         output2 = tensor.dot(input2_, W2)
+    ...         return output1, output2
+    ...
+    ...     def get_dim(self, name):
+    ...         if name == 'input1_':
+    ...             return self.input_dim1
+    ...         if name == 'input2_':
+    ...             return self.input_dim2
+    ...         if name == 'output1':
+    ...             return self.output_dim1
+    ...         if name == 'output2':
+    ...             return self.output_dim2
+    ...         super(ParallelLinear, self).get_dim(name)
+
+You can test the brick as follows:
+
+   >>> input_dim1, input_dim2, output_dim1, output_dim2 = 10, 5, 2, 1
+   >>> batch_size1, batch_size2 = 1, 2
+   >>>
+   >>> x1_mat = 3 * numpy.ones((batch_size1, input_dim1),
+   ...                         dtype=theano.config.floatX)
+   >>> x2_mat = 4 * numpy.ones((batch_size2, input_dim2),
+   ...                         dtype=theano.config.floatX)
+   >>>
+   >>> x1 = theano.tensor.matrix('x1')
+   >>> x2 = theano.tensor.matrix('x2')
+   >>> parallel1 = ParallelLinear(input_dim1, input_dim2, output_dim1,
+   ...                            output_dim2, weights_init=Constant(2))
+   >>> parallel1.initialize()
+   >>> output1, output2 = parallel1.apply(x1, x2)
+   >>>
+   >>> f1 = theano.function([x1, x2], [output1, output2])
+   >>> f1(x1_mat, x2_mat) # doctest: +ELLIPSIS
+   [array([[ 60.,  60.]]...), array([[ 40.],
+          [ 40.]]...)]
+
+One can also create the brick using :class:`Linear` children bricks, which
+
+    >>> class ParallelLinear2(Initializable):
+    ...     def __init__(self, input_dim1, input_dim2, output_dim1, output_dim2,
+    ...                  **kwargs):
+    ...         super(ParallelLinear2, self).__init__(**kwargs)
+    ...         self.linear1 = Linear(input_dim1, output_dim1,
+    ...                               use_bias=False, **kwargs)
+    ...         self.linear2 = Linear(input_dim2, output_dim2,
+    ...                               use_bias=False, **kwargs)
+    ...         self.children = [self.linear1, self.linear2]
+    ...
+    ...     @application(inputs=['input1_', 'input2_'], outputs=['output1',
+    ...         'output2'])
+    ...     def apply(self, input1_, input2_):
+    ...         output1 = self.linear1.apply(input1_)
+    ...         output2 = self.linear2.apply(input2_)
+    ...         return output1, output2
+    ...
+    ...     def get_dim(self, name):
+    ...         if name in ['input1_', 'output1']:
+    ...             return self.linear1.get_dim(name)
+    ...         if name in ['input2_', 'output2']:
+    ...             return self.linear2.get_dim(name)
+    ...         super(ParallelLinear2, self).get_dim(name)
+
+You can test this new version as follows:
```

Comment:
Here we should definitely pay attention to the fact that `weights_init` and `biases_init` are propagated behind the scenes to the linear children, and that this is in effect the main reason to use `Initializable`.